### PR TITLE
[Merged by Bors] - fix: fix simp lemmas about coercion to function

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

File: Mathlib/Algebra/Hom/Equiv/Basic.lean [hunk 1]
```diff
@@ -201,12 +201,16 @@ variable [Mul M] [Mul N] [Mul P] [Mul Q]
 -- Porting note: `to_fun_eq_coe` no longer needed in Lean4
 #noalign mul_equiv.to_fun_eq_coe
 #noalign add_equiv.to_fun_eq_coe
```
Comment (eric-wieser) [inline Mathlib/Algebra/Hom/Equiv/Basic.lean:203]: This note is also wrong, these are needed.

Comment (urkud) [reply Mathlib/Algebra/Hom/Equiv/Basic.lean:203]: The simp normal form is now `coe_toEquiv`

Comment (eric-wieser) [reply Mathlib/Algebra/Hom/Equiv/Basic.lean:203]: Why isn't the simp-normal form `FunLike.coe`?

Comment (urkud) [reply Mathlib/Algebra/Hom/Equiv/Basic.lean:203]: Sorry, I was writing from mobile. I meant that the `simp` lemma we need is `coe_toEquiv`, not `toFun_eq_coe`, becase `e.toFun` is actually `e.toEquiv.toFun`, and `Equiv.toFun_eq_coe` simplifies it to `FunLike.coe e.toEquiv`.

Comment (urkud) [reply Mathlib/Algebra/Hom/Equiv/Basic.lean:203]: Anyway, I'm going to merge this now. We can fix more in another PR.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**